### PR TITLE
[WIP] LLM transformers api for langchain

### PR DESCRIPTION
**Pull request by Litchilitchy**
## Description
LLM transformers api for langchain

### CODE REVIEW

**Comment by hkvision** [comment]
Missing TransformersPipelineLLM tests?

**Comment by hkvision** [comment]
@shane-huang Take a look to see if more tests are needed.

**Comment by Litchilitchy** [comment]
> Missing TransformersPipelineLLM tests?

What's this? Another langchain usage?

**Comment by hkvision** [comment]
> > Missing TransformersPipelineLLM tests?
> 
> What's this? Another langchain usage?

This one? https://github.com/intel-analytics/BigDL/blob/main/python/llm/src/bigdl/llm/langchain/llms/transformerspipelinellm.py

**Comment by Litchilitchy** [comment]
> > > Missing TransformersPipelineLLM tests?
> > 
> > 
> > What's this? Another langchain usage?
> 
> This one? https://github.com/intel-analytics/BigDL/blob/main/python/llm/src/bigdl/llm/langchain/llms/transformerspipelinellm.py

Noted, also add this into README in this PR?

**Comment by Litchilitchy** [comment]
TransformersPipelineLLM added

File: python/llm/test/langchain/test_transformers_api.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+from langchain.vectorstores import Chroma
+
+import pytest
+from unittest import TestCase
+import os
+
+
+
+
+
```
Comment (hkvision) [inline python/llm/test/langchain/test_transformers_api.py:38]: remove redundant lines

File: python/llm/test/langchain/test_transformers_api.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        
+    def test_qa_chain(self):
+        texts = '''
+        AI is a machine’s ability to perform the cognitive functions 
+        we associate with human minds, such as perceiving, reasoning, 
+        learning, interacting with an environment, problem solving,
+            and even exercising creativity. You’ve probably interacted 
+            with AI even if you didn’t realize it—voice assistants like Siri 
+            and Alexa are founded on AI technology, as are some customer 
+            service chatbots that pop up to help you navigate websites.
```
Comment (hkvision) [inline python/llm/test/langchain/test_transformers_api.py:66]: keep the text aligned?

File: python/llm/test/langchain/test_transformers_api.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        #get relavant texts
+        docs = docsearch.get_relevant_documents(query)
+        bigdl_llm = TransformersLLM.from_model_id(model_id=self.auto_model_path, model_kwargs={'trust_remote_code': True})
+        doc_chain = load_qa_chain(bigdl_llm, chain_type="stuff", prompt=QA_PROMPT)
+        output = doc_chain.run(input_documents=docs, question=query)
+        res = "AI" in output
+        self.assertTrue(res)
+        
+if __name__ == '__main__':
+    pytest.main([__file__])
```
Comment (hkvision) [inline python/llm/test/langchain/test_transformers_api.py:84]: new line at the end

**Review by hkvision — APPROVED** [review]

**Pull request merged**